processor/k8sattributesprocessor/e2e_test.go

```diff
@@ -816,7 +818,7 @@ func TestE2E_NamespacedRBACNoPodIP(t *testing.T) {
 				"k8s.container.name":           newExpectedValue(equal, "telemetrygen"),
 				"container.image.name":         newExpectedValue(equal, "ghcr.io/open-telemetry/opentelemetry-collector-contrib/telemetrygen"),
 				"container.image.repo_digests": newExpectedValue(shouldnotexist, ""),
-				"container.image.tag":          newExpectedValue(equal, "latest"),
+				"container.image.tag":          newExpectedValue(shouldnotexist, ""),
```

Comment:
oh good point. i'll add handling for this back. this definitely would've been a regression

Reply:
actually looking at the code, we didn't assume this before. I can add this assertion in easily, though. Since it fits the `docker pull` behavior: https://docs.docker.com/reference/cli/docker/image/pull/#pull-an-image-from-docker-hub

Reply:
Ah right, the test had `ghcr.io/open-telemetry/opentelemetry-collector-contrib/telemetrygen:latest` that's why it was `latest` in the test. 

I'm fine with what was described at https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/36145/files#r1832085419.